spec.html

```diff
@@ -555,7 +557,9 @@ <h1>%Iterator.prototype%.take ( _limit_ )</h1>
         <h1>%Iterator.prototype%.drop ( _limit_ )</h1>
         <emu-alg>
           1. Let _iterated_ be ? GetIteratorDirect(*this* value).
-          1. Let _integerLimit_ be ? ToIntegerOrInfinity(_limit_).
+          1. Let _numLimit_ be ? ToNumber(_limit_).
+          1. If _numLimit_ is *NaN*, throw a *RangeError* exception.
+          1. Let _integerLimit_ be ! ToIntegerOrInfinity(_numLimit_).
```

Comment:
What if `integerLimit` is not in the safe integer range?

For example, in the case of `Number.MAX_SAFE_INTEGER + 2` it will never take a value.

Reply:
This iteration happens on mathematical values, not floats, so the max safe integer for floats doesn't come into play at all here.

Reply:
Reasonable. It seems necessary to think about what should be used in polyfills for such cases.

Reply:
You could use a BigInt, but honestly, you're computer is going to run out of time/memory before it runs into this problem.

Reply:
Iterator helper is one of the solutions for preventing out of memory in such cases and 2**53 is not such a big value that it can't happen for some years of a server work -) BigInt directly will not work for old env. However, I agree that there are many more priority tasks.